core/src/main/java/org/apache/accumulo/core/util/PeekingIterator.java

```diff
@@ -91,4 +92,30 @@ public boolean hasNext() {
     }
     return top != null;
   }
+
+  /**
+   * Advances the underlying iterator looking for a match, up to {@code limit} times.
+   *
+   * @param predicate condition that we are looking for
+   * @param limit number of times that we should look for a match
+   * @return results of the search and number of elements consumed from the underlying iterator
```

Comment:
```suggestion
   * @return true if an elements matched the predicate or false otherwise.  When this returns true the iterator will have a next and what peek() and next() return will match the predicate.